lib/improver/blending/weighted_blend.py

```diff
@@ -401,60 +401,82 @@ def process(self, cube, weights=None):
+                    # Set up aggregator
+                    PERCENTILE_BLEND = (Aggregator(
+                        'weighted_mean',
+                        PercentileBlendingAggregator.aggregate))
+
+                    cube_new = cube_thres.collapsed(self.coord,
+                                                    PERCENTILE_BLEND,
+                                                    arr_percent=percentiles,
+                                                    arr_weights=weights,
+                                                    perc_dim=perc_dim)
+
+                # Else do a simple weighted average
+                elif self.mode == "weighted_mean":
+                    # Equal weights are used as default.
+                    weights_array = None
+                    # Else broadcast the weights to be used by the aggregator.
+                    coord_dim_thres = cube_thres.coord_dims(self.coord)
+                    if weights is not None:
+                        weights_array = (
+                            iris.util.broadcast_to_shape(np.array(weights),
+                                                         cube_thres.shape,
+                                                         coord_dim_thres))
+                    orig_cell_methods = cube_thres.cell_methods
+                    # Calculate the weighted average.
+                    cube_new = cube_thres.collapsed(self.coord,
+                                                    iris.analysis.MEAN,
+                                                    weights=weights_array)
+                    # Update the name of the cell_method created by Iris to
+                    # 'weighted_mean' to be consistent.
+                    new_cell_methods = cube_new.cell_methods
+                    extra_cm = (set(new_cell_methods) -
+                                set(orig_cell_methods)).pop()
+                    add_renamed_cell_method(cube_new,
+                                            extra_cm,
+                                            'weighted_mean')
+
+                # Else use the maximum probability aggregator.
+                elif self.mode == "weighted_maximum":
+                    # Set equal weights if none are provided
+                    if weights is None:
+                        num = len(cube_thres.coord(self.coord).points)
+                        weights = np.ones(num) / float(num)
+                    # Set up aggregator
+                    MAX_PROBABILITY = (Aggregator(
+                        'weighted_maximum',
+                        MaxProbabilityAggregator.aggregate))
+
+                    cube_new = cube_thres.collapsed(self.coord,
+                                                    MAX_PROBABILITY,
+                                                    arr_weights=weights)
+                cubelist.append(cube_new)
+            result = cubelist.merge_cube()
+            if isinstance(cubelist[0].data, np.ma.core.MaskedArray):
```

Comment:
I'm very pleased you found this fix rather than us having to  consider relaxing the CLI acceptance testing which was not ideal.